maps/graph_edges/edges_omo.py

```diff
@@ -110,7 +110,7 @@
     {"from": {"map": "OMO_05", "id": 1},               "to": {"map": "OMO_05", "id": "HiddenYBlockB"}, "reqs": [require(partner="Watt",flag="RF_HiddenBlocksVisible")]}, #* PNK Gourmet Guy Crossing Exit East (North) -> HiddenYBlockB (SuperSoda)
     {"from": {"map": "OMO_05", "id": "HiddenYBlockB"}, "to": {"map": "OMO_05", "id": 1},               "reqs": []}, #* HiddenYBlockB (SuperSoda) -> PNK Gourmet Guy Crossing Exit East (North)
 
-    {"from": {"map": "OMO_05", "id": 0}, "to": {"map": "OMO_05", "id": 0}, "reqs": [require(item="Cake")], "pseudoitems": ["MF_Ch4_GaveCakeToGourmetGuy"]}, #+ Give Cake to Gourmet Guy
+    {"from": {"map": "OMO_05", "id": 0}, "to": {"map": "OMO_05", "id": 0}, "reqs": [require(flag="RF_CanVisitTayceT"),require(item="CakeMix")], "pseudoitems": ["MF_Ch4_GaveCakeToGourmetGuy"]}, #+ Give Cake to Gourmet Guy
```

Comment:
Maybe the requirement should be CakeMix OR Cake? Could even include the other cakes although that would be pretty irelevant considering the low chance of them appearing in most seeds

Reply:
Possibly. Generation wouldn't actually be affected, though, unless Cake was added back into `progression_miscitems` too. (And adding it to progression_miscitems would make it impossible for the Cake to appear in a non-replenishable location such as the vanilla location, I believe.)

Reply:
We have to have one Cake in the pool anyway, as giving TayceT. her FryingPan immediately gives the Cake with no need for ever getting a CakeMix.
Which is also why I'm a bit meh on the logic change for GG, as I'm pretty sure I've never used a CakeMix in casual play for making a Cake, instead I always handed in the one I got in exchange for the FryingPan.
Though I can see changing logic to (Cake OR (CakeMix AND TayceTReachable)).